### PR TITLE
Add TinyCBOR library and test example

### PR DESCRIPTION
**Pull request by acassis**
## Summary
Add TinyCBOR library (initially integrated by Lup Yuen: https://lupyuen.github.io/articles/cbor2 )
## Impact
Users will be able to use CBOR library on NuttX
## Testing
ESP32-DevkitC.

Please ignore code style issue caused by TinyCBOR CamelCase

### CODE REVIEW

File: fsutils/libtinycbor/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/minmea
```
Comment (xiaoxiang781216) [inline fsutils/libtinycbor/.gitignore:1]: why add this

File: examples/cbortest/Makefile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+# TINYCBOR_TEST built-in application info
+
+PROGNAME = $(CONFIG_EXAMPLES_TINYCBOR_TEST_PROGNAME)
+PRIORITY = $(CONFIG_EXAMPLES_TINYCBOR_TEST_PRIORITY)
+STACKSIZE = $(CONFIG_EXAMPLES_TINYCBOR_TEST_STACKSIZE)
+MODULE = $(CONFIG_EXAMPLES_TINYCBOR_TEST)
+
+# TINYCBOR_TEST Example
+CFLAGS += -I$(APPDIR)$(DELIM)fsutils$(DELIM)libtinycbor
```
Comment (xiaoxiang781216) [inline examples/cbortest/Makefile:31]: don't need

File: fsutils/libtinycbor/.gitignore (hunk 1 above)
Comment (acassis) [reply fsutils/libtinycbor/.gitignore:1]: I used minmea was base, so I missed that, let me fix it, thank you!

File: examples/cbortest/Makefile (hunk 2 above)
Comment (acassis) [reply examples/cbortest/Makefile:31]: It is required, without it we get this error:
```
./cbortest_main.c:33:10: fatal error: tinycbor/cbor.h: No such file or directory
 #include "tinycbor/cbor.h"
          ^~~~~~~~~~~~~~~~~
```

**Comment by acassis** [comment]
@xiaoxiang781216 any idea about this error:
```
Error: Unable to resolve action `github/super-linter`, repository not found
```

**Comment by btashton** [comment]
> @xiaoxiang781216 any idea about this error:
> ```
> Error: Unable to resolve action `github/super-linter`, repository not found
> ```

This is an unfortunate breakage here that is being resolved. For now ignore this failure. 
https://github.com/super-linter/super-linter/issues/4150

File: examples/cbortest/Makefile (hunk 2 above)
Comment (xiaoxiang781216) [reply examples/cbortest/Makefile:31]: but you already add it to the search path in Make.defs:
https://github.com/apache/nuttx-apps/pull/1725/files#diff-09f0f42da4d976649c51cc9cba5249554b707d76c1d6aee55c6e7bad598988c9R23

File: fsutils/libtinycbor/Make.defs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+ifneq ($(CONFIG_FSUTILS_TINYCBOR_LIB),)
+CONFIGURED_APPS += $(APPDIR)/fsutils/libtinycbor
+CFLAGS += ${INCDIR_PREFIX}$(APPDIR)/fsutils/tinycbor
```
Comment (xiaoxiang781216) [inline fsutils/libtinycbor/Make.defs:23]:
```suggestion
CFLAGS += ${INCDIR_PREFIX}$(APPDIR)/fsutils/libtinycbor
```

File: examples/cbortest/Makefile (hunk 2 above)
Comment (acassis) [reply examples/cbortest/Makefile:31]: That one is not taking effect, I will remove this one so

File: fsutils/libtinycbor/Make.defs (hunk 3 above)
Comment (acassis) [reply fsutils/libtinycbor/Make.defs:23]: You are right, it was a mistake, it explains why it wasn't working

**Comment by acassis** [comment]
> > @xiaoxiang781216 any idea about this error:
> > ```
> > Error: Unable to resolve action `github/super-linter`, repository not found
> > ```
> 
> This is an unfortunate breakage here that is being resolved. For now ignore this failure. [super-linter/super-linter#4150](https://github.com/super-linter/super-linter/issues/4150)

Thank you Brennan!

File: examples/cbortest/cbortest_main.c [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+int main(int argc, FAR char *argv[])
+{
+  CborError res;
+  CborEncoder encoder;
+  CborEncoder map_encoder;
+  uint8_t output[50];
+  size_t output_len;
+  int i;
+
+  printf("TinyCBOT test: Encoding { \"t\": 1234 }\n");
```
Comment (gustavonihei) [inline examples/cbortest/cbortest_main.c:58]:
```suggestion
  printf("TinyCBOR test: Encoding { \"t\": 1234 }\n");
```

**Review by hartmannathan — COMMENTED** [review]
TinyCBOR is MIT license. Should `config EXAMPLES_TINYCBOR_TEST` have `depends on ALLOW_MIT_COMPONENTS`?

**Comment by acassis** [comment]
> depends on ALLOW_MIT_COMPONENTS

Good point, in fact it needs it to avoid creating a firmware with a different license.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**